src/huggingface_hub/repository.py

```diff
@@ -566,9 +570,12 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
             # checks if repository is initialized in a empty repository or in one with files
             if len(os.listdir(self.local_dir)) == 0:
                 logger.warning(f"Cloning {clean_repo_url} into local empty directory.")
+                prefix = ""
+                if self.skip_lfs:
+                    prefix = "GIT_LFS_SKIP_SMUDGE=1 "
```

Comment:
The `subprocess.run` method takes as argument the `env` parameter, which contains the values for the environment variables. I'd rather we leverage this as it's the most cross-platform way to handle things. 

See the docs here https://docs.python.org/3/library/subprocess.html

Example usage here https://stackoverflow.com/questions/2231227/python-subprocess-popen-with-a-modified-environment